### PR TITLE
refactor(examples): Checkout

### LINKED ISSUE #973: [FEATURE] Checkout - card expiration info should have SfSelect
**Describe how this feature will help us improve**
For better UX we should change SfInput to SfSelect in payment details -> in card info when passing Month and Year of card expiration date:

image

**Is your proposal related to a problem?**
Related with #923

### LINKED ISSUE #974: [BUG] Checkout - adjust Shipping methods to design
**Describe the bug**
We should adjust shipping methods to design. See difference: 
**Expected behavior**
image

**Actual behavior**
What actual happened.
image

Link to Figma https://www.figma.com/file/N0Ct95cSAoODNv7zYS01ng/Storefront-UI-%7C-Design-System

### LINKED ISSUE #977: [FEATURE] Checkout - add SfCollectedProduct list to SfAccordion on mobile
**Describe how this feature will help us improve**
We should add SfCollectedProduct list to order details in SfAccordion on mobile. 

image

### PR DESCRIPTION
**Pull request by martaradziszewska**
# Related issue

Closes #973
Closes #974 
Closes #977
# Scope of work

- changed the way of sending data from each view
- adjusted styles to design

# Screenshots of visual changes

# Checklist

- [x] No commented blocks of code left
- [x] Run tests and docs
- [ ] Self code-reviewed

If applicable:

### CODE REVIEW

**Comment by pspaczek** [comment]
this image on design have size 82px x 112px and probably we should add some spacing here 
Zrzut ekranu 2020-03-31 o 11 29 03

**Comment by pspaczek** [comment]
here we need shadow ;) check design
Zrzut ekranu 2020-03-31 o 11 34 54
🚨 and when you use edit on Personale Detils you need add all data one more time. Expected: all data is save on example page :)

File: packages/vue/src/examples/pages/checkout/Checkout.vue [hunk 1]
```diff
@@ -3,36 +3,38 @@
     <div class="checkout">
       <div class="checkout__main">
         <SfSteps :active="currentStep" @change="updateStep($event)">
-          <SfStep name="Personal Details">
+          <SfStep name="Details">
             <PersonalDetails
-              :order="order"
-              @update:order="updateOrder($event)"
+              v-model="personalDetails"
```
Comment (pspaczek) [inline packages/vue/src/examples/pages/checkout/Checkout.vue]: @martaradziszewska do you think a v-model is the right way? IMO we should use props and $event to be more transparent, WDYT?

File: packages/vue/src/examples/pages/checkout/_internal/PersonalDetails.vue [hunk 2]
```diff
@@ -103,13 +101,16 @@ export default {
...
+    value: {
       type: Object,
       default: () => ({})
+    },
+    buttonName: {
+      type: String,
+      default: ""
     }
   },
   data() {
```
Comment (pspaczek) [inline packages/vue/src/examples/pages/checkout/_internal/PersonalDetails.vue]: @martaradziszewska when you mount this page again you start with empty form :) you must fill this internal data on mount

File: packages/vue/src/examples/pages/checkout/_internal/Shipping.vue [hunk 3]
```diff
@@ -152,13 +173,13 @@ export default {
...
     shippingMethods: {
       type: Array,
       default: () => []
+    },
+    value: {
+      type: Object,
+      default: () => ({})
     }
   },
   data() {
```
Comment (pspaczek) [inline packages/vue/src/examples/pages/checkout/_internal/Shipping.vue]: @martaradziszewska maybe we can improve `countries: []` and use this API ? https://restcountries.eu/rest/v2/all?fields=name

**Comment by pspaczek** [comment]
ok last few small design things
1. it's not a part of accordion 
Zrzut ekranu 2020-03-31 o 11 59 29
2. CVC has 3 numbers, if you need some changes on design's please ask Marcin 
Zrzut ekranu 2020-03-31 o 11 58 12
3. size of mobile images
Zrzut ekranu 2020-03-31 o 11 56 48
4. spacing on mobile for shipping metods 
Zrzut ekranu 2020-03-31 o 11 55 42
5. adjust to design, for sf-button--text on SfButton ;) 
Zrzut ekranu 2020-03-31 o 11 54 11

File: packages/vue/src/examples/pages/checkout/Checkout.vue [hunk 4]
```diff
@@ -222,48 +229,91 @@ export default {
...
   }
 };
 </script>
 <style lang="scss" scoped>
 @import "~@storefront-ui/vue/styles";
+
 #checkout {
   box-sizing: border-box;
-  padding: 0 var(--spacer-big);
+  font-family: var(--font-family-secondary);
```
[the remark this answers is not in the document]
Comment (martaradziszewska) [reply packages/vue/src/examples/pages/checkout/Checkout.vue:287]: e.g. here we use some text inside `<p>` 
should we leave it without font-family? 
image

**Comment by pspaczek** [comment]
@martaradziszewska I added few small fixes

**Pull request merged**